astropy/wcs/wcs.py

```diff
@@ -3119,7 +3119,7 @@ def __repr__(self):
         the `printwcs()` method.
         """
         description = ["WCS Keywords\n", f"Number of WCS axes: {self.naxis!r}"]
-        sfmt = " : " + "".join(["{" + f"{i}" + "!r}  " for i in range(self.naxis)])
+        sfmt = " : " + "".join(["{" f"{i}" "!r}  " for i in range(self.naxis)])
```

Comment:
I do not like proposed change here. Much less clear than `+`. Suggestion:

```suggestion
        sfmt = " : " + ""..join([f"{{{i}!r}} " for i in range(self.naxis)])
```

Reply:
@mcara I think there's an extraneous "."

Reply:
```suggestion
        sfmt = " : " + "".join([f"{{{i}!r}} " for i in range(self.naxis)])
```

Reply:
Yes, I made a mistake.

Reply:
This is another example of a change that conflicts with Ruff rule [ISC001](https://beta.ruff.rs/docs/rules/single-line-implicit-string-concatenation/), but that rule is being addressed in a separate pull request.

Reply:
Whichever is merged first will have to rebase and deal with the change. LGTM.